packages/datadog-instrumentations/src/http2/client.js

```diff
@@ -52,6 +53,9 @@ function createWrapRequest (authority, options) {
 
 function wrapConnect (connect) {
   return function (authority, options) {
+    if (connectChannel.hasSubscribers) {
+      connectChannel.publish({ args: arguments })
```

Comment:
To detect the SSRF vulnerability, we need to check the original parameter of the `connect` method, we don't need the events and the data that the tracer needs.

Reply:
My only problem with this is that the goal of `args` was so that the subscriber doesn't need to know about the structure of the arguments. If we're gonna move this to the subscriber, it should be moved fully. But I would actually prefer if we did the opposite and do both the normalization and denormalization on the publisher side. In any case, any of the two would be fine with me, but let's not have both approaches at the same time.

Reply:
Ok, make sense. The file which understand about the arguments in the instrumentation file. I've changed it to send more normalized args: `{ authority }` 
I changed it also in `http` instrumentation to send `{ ..., originalArgs: { wholeUrl, options }}`

Reply:
The new implementation seems to have the same issue and it's duplicating what is passed to the subscriber. The normalizing/denormalizing of arguments should only happen in once place.

Reply:
I think that you are talking about the changes in `packages/datadog-instrumentations/src/http/client.js`. 

Yes, you are right, we were normalizing it twice. I removed my normalizing method and modify the existing just to add the parameter that we need.